api/versions.go

```diff
@@ -449,8 +449,62 @@ func (self *version2_0) HandleRequest(requestType string, request []string, m ma
 	}
 }
 
+func instCpuStats(last, cur *info.ContainerStats) (*v2.CpuInstStats, error) {
+	if last == nil {
+		return nil, nil
+	}
+	if !cur.Timestamp.After(last.Timestamp) {
+		return nil, fmt.Errorf("container stats move backwards in time")
+	}
+	if len(last.Cpu.Usage.PerCpu) != len(cur.Cpu.Usage.PerCpu) {
+		return nil, fmt.Errorf("different number of cpus")
+	}
+	// Nanoseconds to gain precision and avoid having zero seconds if the
+	// difference between the timestamps is just under a second
+	timeDelta := uint64(cur.Timestamp.Sub(last.Timestamp).Nanoseconds())
+	if timeDelta <= 0 {
```

Comment:
Our timestamps are one seconds apart. If we took samples too close, it might exaggerate a small spike to a large number. Best to ignore tiny bursts or idle cycles.